File: backend/src/main/java/com/financeiro/model/Orcamento.java
package com.financeiro.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "orcamentos", uniqueConstraints = {
    @UniqueConstraint(columnNames = {"familia", "categoria", "mes", "ano"})
})
public class Orcamento {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false, length = 50)
    private String familia;

    @Column(nullable = false, length = 100)
    private String categoria;

    @Column(name = "valor_limite", nullable = false)
    private Double valorLimite;

    @Column(nullable = false)
    private Integer mes;

    @Column(nullable = false)
    private Integer ano;

    @Column(name = "data_criacao", nullable = false, updatable = false)
    private LocalDateTime dataCriacao;

    @Column(name = "data_atualizacao", nullable = false)
    private LocalDateTime dataAtualizacao;
}

File: frontend/src/types/Orcamento.ts
export interface Orcamento {
    id?: number;
    familia: string;
    categoria: string;
    valorLimite: number;
    mes: number;
    ano: number;
    valorGasto?: number;
    percentualUtilizado?: number;
    dataCriacao?: string;
    dataAtualizacao?: string;
}

export interface OrcamentoProgress {
    id: number;
    familia: string;
    categoria: string;
    valorLimite: number;
    valorGasto: number;
    percentualUtilizado: number;
    valorRestante: number;
    status: 'SAFE' | 'WARNING' | 'CRITICAL' | 'EXCEEDED';
    mes: number;
    ano: number;
}
